libraries/extractor/src/main/java/androidx/media3/extractor/text/SubtitleExtractor.java

```diff
@@ -159,6 +164,9 @@ public int read(ExtractorInput input, PositionHolder seekPosition) throws IOExce
       boolean inputFinished = readFromInput(input);
       if (inputFinished) {
         parseAndWriteToOutput();
+        // We do not change the state to STATE_FINISHED, otherwise for segmented webvtt on dash
+        // we will show the subtitle in the first vtt file, and miss all subtitles in next vtt file.
+        if (needsSegmentedSubtitleWorkarounduse) return RESULT_END_OF_INPUT;
```

Comment:
in this case we never set the state to `STATE_FINISHED`? wonder if it could cause some side effect?